tool_collections/bamtools/bamtools_filter/bamtools-filter.xml

```diff
@@ -1,322 +1,346 @@
+                            </sanitizer>
+                        </param>
+                    </when>
+                    <when value="queryBases">
+                        <param name="bam_property_value" type="text" value="ttagggttagg" label="Filter on a sequence motif" help="You can use ! (not) in your expression">
+                            <sanitizer invalid_char="">
+                                <valid initial="string.letters,string.digits"><add value="&gt;"/><add value="&lt;"/><add value="!="/></valid>
+                            </sanitizer>
+                        </param>
+                    </when>
+                    <when value="reference">
+                        <param name="bam_property_value" type="text" value="chr22" label="Filter on the reference name for the read" help="You can use ! (not) in your expression">
+                            <sanitizer invalid_char="">
+                                <valid initial="string.letters,string.digits"><add value="&gt;"/><add value="&lt;"/><add value="!="/></valid>
+                            </sanitizer>
+                        </param>
+                    </when>
+                    <when value="tag">
+                        <param name="bam_property_value" type="text" value="NM:&gt;1" label="Filter on a particular tag" help="You can use &gt;, &lt;, =, and ! (not).  Tag name and its value must be separated by &quot;:&quot;. E.g., to obtain reads with at least one mismatch use &quot;NM:&gt;1&quot;">
+                            <sanitizer invalid_char="">
+                                <valid initial="string.letters,string.digits"><add value="&gt;"/><add value="&lt;"/><add value=":!="/></valid>
+                            </sanitizer>
+                        </param>
+                    </when>
+                </conditional>
+            </repeat>
+        </repeat>
+        <conditional name="rule_configuration">
+            <param name="rules_selector" type="boolean" truevalue="true" falsevalue="false" label="Would you like to set rules?" help="Allows complex logical constructs. See Example 4 below." />
+            <when value="false"/>
+            <when value="true">
+                <param name="rules" type="text" label="Enter rules here" help="This option can only be used with at least two conditions. Read help below (Example 4) to understand how it works." >
+                    <sanitizer invalid_char="">
+                        <valid initial="string.printable"/>
+                    </sanitizer>
+                </param>
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data format="txt" name="out_file2" />
+        <data name="out_file1" format="bam" label="${tool.name} on ${on_string}">
+            <discover_datasets assign_primary_output="True" directory="outputs" pattern="filtered_(?P&lt;designation&gt;.+)\.bam" ext="bam" visible="true" />
```

Comment:
Planemo says: Element 'discover_datasets', attribute 'assign_primary_output': 'True' is not a valid value of the atomic type 'xs:boolean'.

I think you should use lowercase `true`.